### ISSUE TITLE
Love Your Repositories

### ISSUE THREAD

**Issue by sgoguen** [brianberns/LambdaCalculus#1]
Hi Brian!

I just wanted to drop a note and let you know I am thoroughly enjoying looking through your projects.

This weekend, I forked this project and found it very useful for an idea. The idea is:  If you give me an integer, I'll give you a unique lambda function, making the set of lambda functions addressable.  https://github.com/sgoguen/LambdaCalculus

Thank you again!

**Comment by brianberns** [comment]
Hi. Thank you. I'm glad you're enjoying these projects! I certainly learned a lot in creating them (and am still far from an expert on these topics).

Your enhancement sounds interesting, but I'm wondering about this part of the readme:

> I wanted to create a function that effectively maps each nutural number to a unique lambda function so that it creates a bijection between all natural numbers and all lambda functions.

A bijection between natural numbers and lambda functions seems impossible to me, because [such functions are uncountable](https://proofwiki.org/wiki/Natural_Number_Functions_are_Uncountable). Perhaps I'm missing something?

**Comment by sgoguen** [comment]
That's correct.  I'm merely mapping the set of natural numbers to the set of lambda functions that can be constructed with a basic AST, which is really just a serialization problem with an interesting twist.

I agree the set of N -> N is uncountable because, like the vast majority of real numbers, those functions are not representable.

http://www.goodmath.org/blog/2009/05/15/you-cant-write-that-number-in-fact-you-cant-write-most-numbers/
